android/lib/src/main/java/com/marianhello/react/BackgroundGeolocationModule.java

```diff
@@ -433,8 +455,13 @@ private void sendEvent(ReactContext reactContext, String eventName, @Nullable Wr
     }
 
     public boolean hasPermissions() {
-        //TODO: implement
-        return true;
+        final Activity currentActivity = this.getCurrentActivity();
```

Comment:
Right, I was going to get the context from the `currentActivity` assuming it wasn't null, but I guess with the new try/catch statements it's ok to use `getContext()` and have it throw a `NullPointerException`.